src/main/java/com/traveltime/sdk/dto/requests/proto/Transportation.java

```diff
@@ -1,35 +1,197 @@
 package com.traveltime.sdk.dto.requests.proto;
 
+import com.igeolise.traveltime.rabbitmq.requests.RequestsCommon;
 import lombok.AllArgsConstructor;
+import lombok.Builder;
 import lombok.Getter;
 
 public interface Transportation {
-    String getValue();
-    Integer getCode();
+    TransportationType getType();
+
+    RequestsCommon.Transportation getProtoMessage();
+
+    class Modes {
+        public final static Transportation PUBLIC_TRANSPORT = PublicTransport.builder().build();
+        public final static Transportation DRIVING_AND_PUBLIC_TRANSPORT = DrivingAndPublicTransport.builder().build();
+        public final static Transportation WALKING_FERRY = new WalkingFerry();
+        public final static Transportation CYCLING_FERRY = new CyclingFerry();
+        public final static Transportation DRIVING_FERRY = new DrivingFerry();
+        public final static Transportation WALKING = new Walking();
+        public final static Transportation CYCLING = new Cycling();
+        public final static Transportation DRIVING = new Driving();
+    }
+
+    @Builder
+    @Getter
+    class PublicTransport implements Transportation {
+        public final TransportationType type = TransportationType.Types.PUBLIC_TRANSPORT;
+
+        @Builder.Default
+        public TransportationDetails.PublicTransportDetails details =
+            new TransportationDetails.PublicTransportDetails(1800);
+
+        @Override
+        public RequestsCommon.Transportation getProtoMessage() {
+            return RequestsCommon.Transportation
+                .newBuilder()
+                .setPublicTransport(
+                    RequestsCommon.PublicTransportDetails
+                        .newBuilder()
+                        .setWalkingTimeToStation(this.details.walkingTimeToStation)
+                )
+                .setTypeValue(this.type.getCode())
+                .build();
+        }
+    }
+
+    @Builder
+    @Getter
+    class DrivingAndPublicTransport implements Transportation {
+        public final TransportationType type = TransportationType.Types.DRIVING_AND_PUBLIC_TRANSPORT;
+
+        @Builder.Default
+        public final TransportationDetails.DrivingAndPublicTransportDetails details = new TransportationDetails.DrivingAndPublicTransportDetails(
```

Comment:
Since we don't use `getDetails` anymore, maybe we can inline the class here and in `PublicTransport`? Just have `3` fields with a default value each. It will make it easier for the user to update only a single value.